docs/src/network_analysis/odes.md

```diff
@@ -0,0 +1,359 @@
+```@example s1
+Φ = massactionvector(rn)
+```
+
+An important thing to note is this function assumes [combinatoric ratelaws](@ref introduction_to_catalyst_ratelaws), meaning that mass-action products will get rescaled by factorial factors. For instance, note that the mass-action product for the complex `2A + 3B` has a factor of 1/12, corresponding to 1/(2! 3!). This option can be turned off with `combinatoric_ratelaws = false`.
+
+```@example s1
+Φ_2 = massactionvector(rn; combinatoric_ratelaws = false)
+```
+
+Then the reaction rate vector $\mathbf{v}$ can be written as 
+```math
+\mathbf{v}(\mathbf{x}(t)) = K \Phi(\mathbf{x}(t))
+```
+where $K$ is an $R$-by-$C$ matrix called the flux matrix, where $K_{rc}$ is the rate constant of reaction $r$ if $c$ is the index of the substrate complex of reaction $r$, and 0 otherwise. In Catalyst, the API function for $K$ is `fluxmat`: 
+```@example s1
+K = fluxmat(rn)
+```
+
+Since we have that $\mathbf{v} = K\Phi$, we can rewrite the above decompositions as follows: 
+```math
+\begin{align}
+\frac{d\mathbf{x}}{dt} &= N \mathbf{v}(\mathbf{x}(t),t) \\
+&= N K \Phi(\mathbf{x}(t),t) \\
+&= Z B K \Phi(\mathbf{x}(t),t). 
+\end{align}
+```
+
+The final matrix to discuss is the product of $A_k = BK$, which is a $C$-by-$C$ matrix that turns out to be exactly the negative of the [graph Laplacian](https://en.wikipedia.org/wiki/Laplacian_matrix) of the weighted graph whose nodes are reaction complexes and whose edges represent reactions, weighted by the rate constants. The API function for $A_k$ is the `laplacianmat`: 
+```@example s1
+A_k = laplacianmat(rn)
+```
+We can check that 
+```@example s1
+isequal(A_k, B * K)
+```
+
+Note that we have used `isequal` instead of `==` here because `laplacianmat`
+returns a `Matrix{Num}`, since some of its entries are symbolic rate constants.
+
+In sum, we have that
+```math
+\begin{align}
+\frac{d\mathbf{x}}{dt} &= N \mathbf{v}(\mathbf{x}(t),t) \\
+&= N K \Phi(\mathbf{x}(t),t) \\
+&= Z B K \Phi(\mathbf{x}(t),t). 
+&= Z A_k \Phi(\mathbf{x}(t),t). 
+\end{align}
+```
+
+All three of the objects introduced in this section (the flux matrix, mass-action vector, Laplacian matrix) will return symbolic outputs by default, but can be made to return numerical outputs if values are specified. 
+For example, `massactionvector` will return a numerical output if a set of species concentrations is supplied using a dictionary, tuple, or vector of Symbol-value pairs.
+```@example s1
+concmap = Dict([:A => 3., :B => 5., :C => 2.4, :D => 1.5])
+massactionvector(rn, concmap)
+```
+
+`fluxmat` and `laplacianmat` will return numeric matrices if a set of rate constants and other aprameters are supplied the same way.
+```@example s1
+parammap = Dict([:k => 12., :b => 8.])
+fluxmat(rn, parammap)
+```
+
+```@example s1
+laplacianmat(rn, parammap)
+```
+
+# Symbolic ODE functions
```

Comment:
I think so far (not sure by chance or planning, but I stick to it these days) we only use `# Title` for the first header of each doc page. Then subsequent ones uses `## Sub heading` and `### Subsub heading` and such.